File: srv/scheduling/common/JobSchedulingError.js
"use strict";

const BaseError = require("../../common/BaseError");

class JobSchedulingError extends BaseError {
  static accessOnlyViaParent() {
    return new JobSchedulingError("accessOnlyViaParent", {
      args: [],
      httpStatus: 400,
    });
  }

  static jobNotFound(ID) {
    return new JobSchedulingError("jobNotFound", {
      args: [ID],
      httpStatus: 404,
    });
  }

  static statusValueMissing() {
    return new JobSchedulingError("statusValueMissing", {
      args: [],
      httpStatus: 400,
    });
  }

  static invalidJobStatus(status) {
    return new JobSchedulingError("invalidJobStatus", {
      args: [status],
      httpStatus: 400,
    });
  }

  static statusTransitionNotAllowed(statusBefore, statusAfter) {
    return new JobSchedulingError("statusTransitionNotAllowed", {
      args: [statusBefore, statusAfter],
      httpStatus: 400,
    });
  }

  static invalidOption(value, option) {
    return new JobSchedulingError("invalidOption", {
      args: [value, option],
      httpStatus: 400,
    });
  }

  static jobDefinitionNotFound(name) {
    return new JobSchedulingError("jobDefinitionNotFound", {
      args: [name],
      httpStatus: 400,
    });
  }

  static referenceIDMissing() {
    return new JobSchedulingError("referenceIDMissing", {
      args: [],
      httpStatus: 400,
    });
  }

  static referenceIDNoUUID(referenceID) {
    return new JobSchedulingError("referenceIDNoUUID", {
      args: [referenceID],
      httpStatus: 400,
    });
  }

  static jobResultsReadOnly() {
    return new JobSchedulingError("jobResultsReadOnly", {
      args: [],
      httpStatus: 400,
    });
  }

  static startDateTimeNotSupported(name) {
    return new JobSchedulingError("startDateTimeNotSupported", {
      args: [name],
      httpStatus: 400,
    });
  }

  static jobParameterNameMissing() {
    return new JobSchedulingError("jobParameterNameMissing", {
      args: [],
      httpStatus: 400,
    });
  }

  static jobParameterNotKnown(name) {
    return new JobSchedulingError("jobParameterNotKnown", {
      args: [name],
      httpStatus: 400,
    });
  }

  static jobParameterRequired(name) {
    return new JobSchedulingError("jobParameterRequired", {
      args: [name],
      httpStatus: 400,
    });
  }

  static jobParameterReadOnly(name) {
    return new JobSchedulingError("jobParameterReadOnly", {
      args: [name],
      httpStatus: 400,
    });
  }

  static jobParameterValueRequired(name) {
    return new JobSchedulingError("jobParameterValueRequired", {
      args: [name],
      httpStatus: 400,
    });
  }

  static jobParameterValueInvalidType(value, name, type) {
    return new JobSchedulingError("jobParameterValueInvalidType", {
      args: [value, name, type],
      httpStatus: 400,
    });
  }

  static jobCannotBeCanceled(status) {
    return new JobSchedulingError("jobCannotBeCanceled", {
      args: [status],
      httpStatus: 400,
    });
  }

  static jobResultNotFound(ID) {
    return new JobSchedulingError("jobResultNotFound", {
      args: [ID],
      httpStatus: 404,
    });
  }

  static resultNameMissing() {
    return new JobSchedulingError("resultNameMissing", {
      args: [],
      httpStatus: 400,
    });
  }

  static resultTypeMissing() {
    return new JobSchedulingError("resultTypeMissing", {
      args: [],
      httpStatus: 400,
    });
  }

  static invalidResultType(resultType) {
    return new JobSchedulingError("invalidResultType", {
      args: [resultType],
      httpStatus: 400,
    });
  }

  static linkMissing(resultType) {
    return new JobSchedulingError("linkMissing", {
      args: [resultType],
      httpStatus: 400,
    });
  }

  static mimeTypeMissing(resultType) {
    return new JobSchedulingError("mimeTypeMissing", {
      args: [resultType],
      httpStatus: 400,
    });
  }

  static filenameMissing(resultType) {
    return new JobSchedulingError("filenameMissing", {
      args: [resultType],
      httpStatus: 400,
    });
  }

  static dataMissing(resultType) {
    return new JobSchedulingError("dataMissing", {
      args: [resultType],
      httpStatus: 400,
    });
  }

  static messagesMissing(resultType) {
    return new JobSchedulingError("messagesMissing", {
      args: [resultType],
      httpStatus: 400,
    });
  }

  static codeMissing() {
    return new JobSchedulingError("codeMissing", {
      args: [],
      httpStatus: 400,
    });
  }

  static textMissing() {
    return new JobSchedulingError("textMissing", {
      args: [],
      httpStatus: 400,
    });
  }

  static localeMissing() {
    return new JobSchedulingError("localeMissing", {
      args: [],
      httpStatus: 400,
    });
  }

  static invalidLocale(locale) {
    return new JobSchedulingError("invalidLocale", {
      args: [locale],
      httpStatus: 400,
    });
  }

  static severityMissing() {
    return new JobSchedulingError("severityMissing", {
      args: [],
      httpStatus: 400,
    });
  }

  static invalidMessageSeverity(messageSeverity) {
    return new JobSchedulingError("invalidMessageSeverity", {
      args: [messageSeverity],
      httpStatus: 400,
    });
  }

  static linkNotAllowed(resultType) {
    return new JobSchedulingError("linkNotAllowed", {
      args: [resultType],
      httpStatus: 400,
    });
  }

  static mimeTypeNotAllowed(resultType) {
    return new JobSchedulingError("mimeTypeNotAllowed", {
      args: [resultType],
      httpStatus: 400,
    });
  }

  static filenameNotAllowed(resultType) {
    return new JobSchedulingError("filenameNotAllowed", {
      args: [resultType],
      httpStatus: 400,
    });
  }

  static dataNotAllowed(resultType) {
    return new JobSchedulingError("dataNotAllowed", {
      args: [resultType],
      httpStatus: 400,
    });
  }

  static messagesNotAllowed(resultType) {
    return new JobSchedulingError("messagesNotAllowed", {
      args: [resultType],
      httpStatus: 400,
    });
  }
}

module.exports = JobSchedulingError;

File: java/project/srv/src/main/java/scheduling/common/JobSchedulingError.java
package scheduling.common;

import com.sap.cds.services.ErrorStatus;
import com.sap.cds.services.ErrorStatuses;
import com.sap.cds.services.ServiceException;

public class JobSchedulingError extends ServiceException {

    public JobSchedulingError(ErrorStatus errorStatus, String messageOrKey, Object... args) {
        super(errorStatus, messageOrKey, args);
    }

    public static JobSchedulingError accessOnlyViaParent() {
        return new JobSchedulingError(ErrorStatuses.BAD_REQUEST, "accessOnlyViaParent");
    }

    public static JobSchedulingError jobNotFound(String ID) {
        return new JobSchedulingError(ErrorStatuses.NOT_FOUND, "jobNotFound", ID);
    }

    public static JobSchedulingError statusValueMissing() {
        return new JobSchedulingError(ErrorStatuses.BAD_REQUEST, "statusValueMissing");
    }

    public static JobSchedulingError invalidJobStatus(String status) {
        return new JobSchedulingError(ErrorStatuses.BAD_REQUEST, "invalidJobStatus", status);
    }

    public static JobSchedulingError statusTransitionNotAllowed(String statusBefore, String statusAfter) {
        return new JobSchedulingError(ErrorStatuses.BAD_REQUEST, "statusTransitionNotAllowed", statusBefore, statusAfter);
    }

    public static JobSchedulingError invalidOption(Object value, String option) {
        return new JobSchedulingError(ErrorStatuses.BAD_REQUEST, "invalidOption", value, option);
    }

    public static JobSchedulingError jobDefinitionNotFound(String name) {
        return new JobSchedulingError(ErrorStatuses.NOT_FOUND, "jobDefinitionNotFound", name);
    }

    public static JobSchedulingError referenceIDMissing() {
        return new JobSchedulingError(ErrorStatuses.BAD_REQUEST, "referenceIDMissing");
    }

    public static JobSchedulingError referenceIDNoUUID(String referenceID) {
        return new JobSchedulingError(ErrorStatuses.BAD_REQUEST, "referenceIDNoUUID", referenceID);
    }

    public static JobSchedulingError jobResultsReadOnly() {
        return new JobSchedulingError(ErrorStatuses.BAD_REQUEST, "jobResultsReadOnly");
    }

    public static JobSchedulingError startDateTimeNotSupported(String name) {
        return new JobSchedulingError(ErrorStatuses.BAD_REQUEST, "startDateTimeNotSupported", name);
    }

    public static JobSchedulingError jobParameterNameMissing() {
        return new JobSchedulingError(ErrorStatuses.BAD_REQUEST, "jobParameterNameMissing");
    }

    public static JobSchedulingError jobParameterNotKnown(String name) {
        return new JobSchedulingError(ErrorStatuses.BAD_REQUEST, "jobParameterNotKnown", name);
    }

    public static JobSchedulingError jobParameterRequired(String name) {
        return new JobSchedulingError(ErrorStatuses.BAD_REQUEST, "jobParameterRequired", name);
    }

    public static JobSchedulingError jobParameterReadOnly(String name) {
        return new JobSchedulingError(ErrorStatuses.BAD_REQUEST, "jobParameterReadOnly", name);
    }

    public static JobSchedulingError jobParameterValueRequired(String name) {
        return new JobSchedulingError(ErrorStatuses.BAD_REQUEST, "jobParameterValueRequired", name);
    }

    public static JobSchedulingError jobParameterValueInvalidType(Object value, String name, String type) {
        return new JobSchedulingError(ErrorStatuses.BAD_REQUEST, "jobParameterValueInvalidType", value, name, type);
    }

    public static JobSchedulingError jobCannotBeCanceled(String status) {
        return new JobSchedulingError(ErrorStatuses.BAD_REQUEST, "jobCannotBeCanceled", status);
    }

    public static JobSchedulingError jobResultNotFound(String ID) {
        return new JobSchedulingError(ErrorStatuses.BAD_REQUEST, "jobResultNotFound", ID);
    }

    public static JobSchedulingError resultNameMissing() {
        return new JobSchedulingError(ErrorStatuses.BAD_REQUEST, "resultNameMissing");
    }

    public static JobSchedulingError resultTypeMissing() {
        return new JobSchedulingError(ErrorStatuses.BAD_REQUEST, "resultTypeMissing");
    }

    public static JobSchedulingError invalidResultType(String resultType) {
        return new JobSchedulingError(ErrorStatuses.BAD_REQUEST, "invalidResultType", resultType);
    }

    public static JobSchedulingError linkMissing(String resultType) {
        return new JobSchedulingError(ErrorStatuses.BAD_REQUEST, "linkMissing", resultType);
    }

    public static JobSchedulingError mimeTypeMissing(String resultType) {
        return new JobSchedulingError(ErrorStatuses.BAD_REQUEST, "mimeTypeMissing", resultType);
    }

    public static JobSchedulingError filenameMissing(String resultType) {
        return new JobSchedulingError(ErrorStatuses.BAD_REQUEST, "filenameMissing", resultType);
    }

    public static JobSchedulingError dataMissing(String resultType) {
        return new JobSchedulingError(ErrorStatuses.BAD_REQUEST, "dataMissing", resultType);
    }

    public static JobSchedulingError messagesMissing(String resultType) {
        return new JobSchedulingError(ErrorStatuses.BAD_REQUEST, "messagesMissing", resultType);
    }

    public static JobSchedulingError codeMissing() {
        return new JobSchedulingError(ErrorStatuses.BAD_REQUEST, "codeMissing");
    }

    public static JobSchedulingError textMissing() {
        return new JobSchedulingError(ErrorStatuses.BAD_REQUEST, "textMissing");
    }

    public static JobSchedulingError localeMissing() {
        return new JobSchedulingError(ErrorStatuses.BAD_REQUEST, "localeMissing");
    }

    public static JobSchedulingError invalidLocale(String locale) {
        return new JobSchedulingError(ErrorStatuses.BAD_REQUEST, "invalidLocale", locale);
    }

    public static JobSchedulingError severityMissing() {
        return new JobSchedulingError(ErrorStatuses.BAD_REQUEST, "severityMissing");
    }

    public static JobSchedulingError invalidMessageSeverity(String messageSeverity) {
        return new JobSchedulingError(ErrorStatuses.BAD_REQUEST, "invalidMessageSeverity", messageSeverity);
    }

    public static JobSchedulingError linkNotAllowed(String resultType) {
        return new JobSchedulingError(ErrorStatuses.BAD_REQUEST, "linkNotAllowed", resultType);
    }

    public static JobSchedulingError mimeTypeNotAllowed(String resultType) {
        return new JobSchedulingError(ErrorStatuses.BAD_REQUEST, "mimeTypeNotAllowed", resultType);
    }

    public static JobSchedulingError filenameNotAllowed(String resultType) {
        return new JobSchedulingError(ErrorStatuses.BAD_REQUEST, "filenameNotAllowed", resultType);
    }

    public static JobSchedulingError dataNotAllowed(String resultType) {
        return new JobSchedulingError(ErrorStatuses.BAD_REQUEST, "dataNotAllowed", resultType);
    }

    public static JobSchedulingError messagesNotAllowed(String resultType) {
        return new JobSchedulingError(ErrorStatuses.BAD_REQUEST, "messagesNotAllowed", resultType);
    }

}
